internal/show/versions.go

```diff
@@ -67,19 +94,119 @@ func printDeploymentVersions(clusterInfo *cluster.Info, printer *table.Printer,
 		}
 
 		version, repository := images.ParseOperatorImage(deployment.Spec.Template.Spec.Containers[0].Image)
-		printer.Add(component, repository, version)
+
+		runningVersion, err := getVersionForComponent(clusterInfo, component,
+			labels.SelectorFromSet(deployment.Spec.Selector.MatchLabels).String())
+		if err != nil {
+			return err
+		}
+
+		printer.Add(component, repository, version, runningVersion)
 	}
 
 	return nil
 }
 
+func getVersionForComponent(clusterInfo *cluster.Info, component, labelSelector string) (string, error) {
+	podsClient := clusterInfo.ClientProducer.ForKubernetes().CoreV1().Pods(constants.OperatorNamespace)
+	podList, err := podsClient.List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+
+	if err != nil || len(podList.Items) < 1 {
+		return "", errors.Wrapf(err, "Failed to find pods for component %s", component)
+	}
+
+	// Try all pods
+	for i := range podList.Items {
+		pod := podList.Items[i]
+		podVersion, _ := getVersionFromPodBinary(&pod, clusterInfo, component)
+
+		if podVersion != "" {
+			return podVersion, nil
+		}
+
+		podVersion = getVersionFromPodLogs(&pod, podsClient, component)
+
+		if podVersion == "" {
+			continue
+		}
+
+		return podVersion, nil
```

Comment:
To match the earlier construct,

```suggestion
		if podVersion != "" {
			return podVersion, nil
		}
```

Reply:
Tricky part here is, if we return error `show version` will fail and not show rest of content. Also, in case of multiple pods, first one may have error but next one may still work. From user perspective, it would make sense to not throw error but return "Unavailable" or something like that.

Error would've been useful if subctl had its own log other than the user visible output. Error helps figure out why it failed, but that can be figured out from `gather` output. While error was helpful to me during dev it doesn't seem too helpful for actual users.

Reply:
A return error should indicate a non-recoverable failure attempting to determine the version, eg K8s API server not available.  If we can't determine a version not due to an actual failure then we shouldn't return an error, instead return "N/A".